src/main/java/org/codeforamerica/messaging/config/SecurityConfiguration.java

```diff
@@ -1,21 +1,36 @@
 package org.codeforamerica.messaging.config;
 
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authorization.AuthorizationDecision;
+import org.springframework.security.authorization.AuthorizationManager;
+import org.springframework.security.authorization.AuthorizationManagers;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
 import org.springframework.security.config.http.SessionCreationPolicy;
 import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.access.intercept.RequestAuthorizationContext;
+import org.springframework.security.web.util.matcher.IpAddressMatcher;
 
+import java.util.Arrays;
+
+import static org.springframework.security.authorization.AuthenticatedAuthorizationManager.authenticated;
 import static org.springframework.security.config.Customizer.withDefaults;
 
 @Configuration
 public class SecurityConfiguration {
+
+    @Value("${allowed-ip-addresses}")
+    String allowedIpAddresses;
+
     @Bean
     public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
         http
                 .csrf(AbstractHttpConfigurer::disable)
                 .authorizeHttpRequests(auth -> auth
+                        .requestMatchers("/api/v1/**").access(AuthorizationManagers.allOf(authenticated(), checkForAllowedIp()))
                         .requestMatchers("/api/v1/**").authenticated()
```

Comment:
Is this line duplicating the line above? Can it be removed?

Reply:
Yes, I left it in accidentally (FWIW it doesn't get triggered - only the first line that matches a given url is actually used, afaict)